### DISCUSSION TITLE
[归档] 重写为浏览器扩展的计划以及进度

### DISCUSSION THREAD

**Discussion by beijixiaohu** [beijixiaohu/OJBetter#165] [Ideas]
该插件将重写为浏览器扩展，不再依赖脚本管理器，以应对 Chrome 政策的未来变化，并便于功能实现和代码组织。

新的扩展将采用现代前端开发的主流工具链，包括 Vue 3、Vite、Pinia、TypeScript、Tailwind CSS、ESLint 和 Prettier 等。

这将使代码更易于扩展、通用且可读，同时还会添加一些新的实用功能，这些功能在现有的 Tampermonkey 脚本中难以实现。

这是一个相当复杂的项目，作者仍需要一些时间进行开发，预计将在**年底**发布。

尽管道路充满未知，但已经取得了一些令人振奋的成果，敬请期待！

**Comment by beijixiaohu** [comment]
如有任何想法或建议，欢迎跟帖

**Comment by funcdfs** [comment]
伟大！无需多言！

- 感觉设置按钮可以做成 沉浸式翻译 扩展那种悬浮式的,
![image](https://github.com/beijixiaohu/OJBetter/assets/56250171/42f8b1a7-ed7d-4800-b964-86ae8daffc17)
 - 翻译之后的字体大小最好可以手动控制px。（因为现在想要改字体大小的话直接手动改脚本了，如果是 Chrome 扩展那改脚本的方法就 GG了，字体大小不自由就很难受了）

**Comment by beijixiaohu** [comment]
> 伟大！无需多言！
> 
> * 感觉设置按钮可以做成 沉浸式翻译 扩展那种悬浮式的,
>   ![image](https://private-user-images.githubusercontent.com/56250171/336353118-42f8b1a7-ed7d-4800-b964-86ae8daffc17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0OTAxNTksIm5iZiI6MTcxNzQ4OTg1OSwicGF0aCI6Ii81NjI1MDE3MS8zMzYzNTMxMTgtNDJmOGIxYTctZWQ3ZC00ODAwLWI5NjQtODZhZThkYWZmYzE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA0VDA4MzA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlNjkwZThkNjg5ZDc2ZTYxMDE2NWE0NDIxN2Y5NzA3M2VjYTFjMGM1NDgxYWEzMWUzMzgyNDhmMTVmOWZiMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qsGaxWvQyC3Nv0rV-DesO_Iy06nVQzAcs8uKduw0blo)
> * 翻译之后的字体大小最好可以手动控制px。（因为现在想要改字体大小的话直接手动改脚本了，如果是 Chrome 扩展那改脚本的方法就 GG了，字体大小不自由就很难受了）

感谢建议，会提供一个详细的设置页面，右键-选项可以打开

![image](https://github.com/beijixiaohu/OJBetter/assets/60805870/d064ea8a-837f-47bc-90d0-1f6d5bd9ff05)

左键点击扩展图标会有一个常用的选项菜单，目前规划大概长这样：

![image](https://github.com/beijixiaohu/OJBetter/assets/60805870/9ac17e02-1090-4cf8-a123-ef63ed6a0806)

也会加一个像沉浸式翻译那样的悬浮球

样式方面，对于各个组件，都会预置几个样式，可以直接切换，此外也可以自定义样式，就像划词翻译这个扩展那样

![image](https://github.com/beijixiaohu/OJBetter/assets/60805870/877ccdd4-12d7-4d9e-8e39-b1eb1ea052b8)

**Comment by wrk-123** [comment]
在浏览器扩展正式发布后，原有的脚本代码是直接不再维护/保留，维护并同步更新一段时间/永久保留，维护并同步更新？

**Comment by beijixiaohu** [comment]
> 在浏览器扩展正式发布后，原有的脚本代码是直接不再维护/保留，维护并同步更新一段时间/永久保留，维护并同步更新？

仅维护，不再添加新的功能，脚本会一直保留的

**Comment by ZZC2012** [comment]
能不能把插件的ZIP文件发给我，我来用用。

**Comment by ZZC2012** [comment]
其实我也是个开发者，我在做我的世界2D版，最近刚调完，可是还是有好多BUG。。。。。。

**Comment by beijixiaohu** [comment]
> 能不能把插件的ZIP文件发给我，我来用用。

哈哈哈好，但是现在啥也没有，功能都还没写，（虽然已经几百个文件了，

摆了两个月了，现在还是两个月前的进度，这两天又有了兴致，~~打开改进了下主题切换的性能，配置读取的性能还没改~~

_刚发现还改出了bug，开发环境中主题切换没问题，打包后作为扩展运行时主题切换有问题……_

[build.zip](https://github.com/user-attachments/files/16385954/build.zip)

**Reply by beijixiaohu** [reply]
修好了🎉

- 修复主题切换时出现错误的问题

  > 原因是 watchEffect 会随着组件销毁而销毁，即使是在闭包中或者在pinia的store中。使用 `effectScope(true)` 可以创建分离的作用域，来保持不销毁，（又学到了）
- 改进了配置读取的速度（也是缓存大法）

现在基本感觉不到明显的卡顿和延迟了

[build.zip](https://github.com/user-attachments/files/16392676/build.zip)

**Comment by beijixiaohu** [comment]
@ZZC2012 有兴趣的话，可以到时候基本做好了艾特你来测试哈哈哈，这个项目还是有很多有意思的技术创新点的

**Comment by ZZC2012** [comment]
栓Q

**Comment by ZZC2012** [comment]
扎心了老铁：
加载扩展程序时出错
无法加载以下来源的扩展程序：C:\Users\Root\Download\build.zip.Could not unzip extension for install.

**Comment by beijixiaohu** [comment]
> 扎心了老铁： 加载扩展程序时出错 无法加载以下来源的扩展程序：C:\Users\Root\Download\build.zip.Could not unzip extension for install.

直接先解压，然后点这儿
<img width="142" alt="Snipaste_2024-07-26_13-47-56" src="https://github.com/user-attachments/assets/b2cf4bfa-20c8-4cfc-b408-76f0305002c4">
